# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/blink/PeriodicWave.cpp
# Commit: e519654e5d8b
# Full Hash: e519654e5d8b7e0dc682c316fa0d16318b962765
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-28 04:30:28
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Scale the inverse FFT result using tx instead of pre- or post-processing the data. r=karlt
#   
#   This skips a copy at the expense of having to do a `const_cast`.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202434
# ==============================================================================

diff -r 9b17ac4800c6 -r e519654e5d8b dom/media/webaudio/blink/PeriodicWave.cpp
--- a/dom/media/webaudio/blink/PeriodicWave.cpp	Tue Feb 27 12:40:23 2024 +0000
+++ b/dom/media/webaudio/blink/PeriodicWave.cpp	Tue Feb 27 12:40:24 2024 +0000
@@ -266,7 +266,7 @@
 
   // Apply an inverse FFT to generate the time-domain table data.
   float* data = m_bandLimitedTables[rangeIndex]->Elements();
-  frame.GetInverseWithoutScaling(data);
+  frame.GetInverse(data);
 
   // For the first range (which has the highest power), calculate
   // its peak value then compute normalization scale.
